integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -2075,4 +2081,400 @@ describe("Flexible sync", function () {
+              name: "Heavy",
+              age: 36,
+              data: new ArrayBuffer(1_000_000),
+            });
+          });
+
+          await realm.syncSession?.uploadAllLocalChanges();
+
+          expect(callback.callCount).to.be.equal(oldCallCount);
+        });
+      });
+
+      describe(`with ProgressDirection.Download`, function () {
+        it("should not call the callback when there is nothing to download", async function (this: RealmContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+
+          realm.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Download,
+            ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          await this.realm.syncSession?.uploadAllLocalChanges();
+          await this.realm.syncSession?.downloadAllServerChanges();
+
+          // TODO: This callback should not be called at this stage but seems flakey
+          // and gets called with 1.0 at times, likely because of a race condition.
+          expect(callback.notCalled || callback.calledWith(1.0)).is.true;
+        });
+
+        it("should be called multiple times with different values during downloads", async function (this: RealmContext &
+          AppContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+          const realm2 = await Realm.open({
+            schema: [Person, Dog],
+            sync: {
+              flexible: true,
+              user: await this.app.logIn(Credentials.anonymous(false)),
+              newRealmFileBehavior: {
+                type: OpenRealmBehaviorType.OpenImmediately,
+              },
+            },
+          });
+
+          realm2.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Download,
+            Realm.ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          const subscription = await realm.objects(Person).subscribe();
+
+          realm.write(() => {
+            realm.create(Person, {
+              _id: new BSON.ObjectId(),
+              name: "Heavy",
+              age: 36,
+              data: new ArrayBuffer(1_000_000),
+            });
+          });
+
+          await realm.syncSession?.uploadAllLocalChanges();
+          const subscription2 = await realm2.objects(Person).subscribe();
+          await realm2.syncSession?.downloadAllServerChanges();
+
+          expect(callback.args.find(([estimate]) => estimate < 1)).to.not.be.undefined;
+          expect(callback.withArgs(1.0).callCount).is.greaterThanOrEqual(2);
+
+          subscription.unsubscribe();
+          subscription2.unsubscribe();
+          realm2.close();
```

Comment:
If using the `openRealmBeforeEach()` hook, I think we could remove these explicit `close()` calls, even if it's on a different realm. The hook will also call `Realm.clearTestState()` after each test which closes and deletes all realm files in the default directory.